internal/hook/item_hook_handler.go

```diff
@@ -76,6 +79,79 @@ type ItemHookHandler interface {
 	) error
 }
 
+// ItemRestoreHookHandler invokes restore hooks for an item
+type ItemRestoreHookHandler interface {
+	// HandleRestoreHooks runs the restore hooks for an item.
+	// If the item is a pod hooks are chosen to be run as follows:
+	// If the pod has the appropriate annotations specifying the hook action, then hooks from the annotation are run
+	// Otherwise, the supplied ResourceRestoreHooks are applied.
+	HandleRestoreHooks(
+		log logrus.FieldLogger,
+		groupResource schema.GroupResource,
+		obj runtime.Unstructured,
+		rh []ResourceRestoreHook,
+	) (runtime.Unstructured, error)
+}
+
+// InitContainerRestoreHookHandler is the restore hook handler to add init containers to restored pods.
+type InitContainerRestoreHookHandler struct{}
+
+func (i *InitContainerRestoreHookHandler) HandleRestoreHooks(
+	log logrus.FieldLogger,
+	groupResource schema.GroupResource,
+	obj runtime.Unstructured,
+	resrouceRestoreHooks []ResourceRestoreHook,
+) (runtime.Unstructured, error) {
+
+	// We only support hooks on pods right now
+	if groupResource != kuberesource.Pods {
+		return nil, nil
+	}
+
+	metadata, err := meta.Accessor(obj)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get a metadata accessor")
+	}
+	pod := new(corev1api.Pod)
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.UnstructuredContent(), pod); err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	initContainers := []corev1api.Container{}
```

Comment:
A comment here as to why we're doing this would be helpful for future readers.

Reply:
added.